sw/host/rom_ext_image_tools/signer/src/main.rs

```diff
@@ -76,10 +82,43 @@ fn _parse_hex_str(hex_str: &str) -> Result<Vec<u8>> {
 
 /// Updates the manifest of an image.
 // TODO: This function must use a public key.
-fn update_image_manifest(image: &mut Image, key: &PrivateKey) -> Result<()> {
+fn update_image_manifest(
+    image: &mut Image,
+    key: &PrivateKey,
+    elf: &ElfFile32,
+) -> Result<()> {
+    let manifest_addr = u32::try_from(
+        elf.section_by_name(".manifest")
+            .context("Could not find the `.manifest` section.")?
+            .address(),
+    )?;
+
     *image.manifest = Manifest {
         identifier: 0x4552544f,
         image_length: u32::try_from(image.bytes().len())?,
+        entry_point: {
+            let addr = u32::try_from(elf.entry())?;
+            addr.checked_sub(manifest_addr).context("Overflow")?
+        },
+        code_start: {
```

Comment:
Do you mean a single section or symbols to mark the start and the end of the executable region? @cfrantz's comments in #6333 seem to suggest we may want to have a separate section for shutdown anyway. It probably would not make a big difference in terms of complexity (still need the start and the end offset) but would be easier to maintain.

Reply:
Sorry, yes, just the single `.text` section containing all executable code bit. You're also right of course that the change would need to be applied to the `ROM_EXT` linker script rather than the mask ROM linker script that #6333 modifies.

It's probably better to keep separate sections, it just seems a bit fragile.